File: room3/integration-tests/kotlintestapp/src/androidTestWithKspGenJava/java/androidx/room3/integration/kotlintestapp/test/PreKmpDatabase_AutoMigration_1_2_Impl.java
package androidx.room3.integration.kotlintestapp.test;

import androidx.annotation.NonNull;
import androidx.room3.migration.AutoMigrationSpec;
import androidx.room3.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;
import java.lang.Override;
import java.lang.SuppressWarnings;
import javax.annotation.processing.Generated;

@Generated("androidx.room3.RoomProcessor")
@SuppressWarnings({"unchecked", "deprecation"})
final class PreKmpDatabase_AutoMigration_1_2_Impl extends Migration {
    private final AutoMigrationSpec callback;

    public PreKmpDatabase_AutoMigration_1_2_Impl(@NonNull final AutoMigrationSpec callback) {
        super(1, 2);
        this.callback = callback;
    }

    @Override
    public void migrate(@NonNull final SupportSQLiteDatabase db) {
        db.execSQL("DROP TABLE `DeletedEntity`");
        callback.onPostMigrate(db);
    }
}

File: room3/integration-tests/kotlintestapp/src/androidTestWithKspGenKotlin/java/androidx/room3/integration/kotlintestapp/test/PreKmpDatabase_AutoMigration_1_2_Impl.kt
package androidx.room3.integration.kotlintestapp.test

import androidx.room3.migration.AutoMigrationSpec
import androidx.room3.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase
import javax.`annotation`.processing.Generated
import kotlin.Suppress

@Generated(value = ["androidx.room3.RoomProcessor"])
@Suppress(names = ["UNCHECKED_CAST", "DEPRECATION", "REDUNDANT_PROJECTION"])
internal class PreKmpDatabase_AutoMigration_1_2_Impl : Migration {
    private val callback: AutoMigrationSpec

    public constructor(callback: AutoMigrationSpec) : super(1, 2) {
        this.callback = callback
    }

    public override fun migrate(db: SupportSQLiteDatabase) {
        db.execSQL("DROP TABLE `DeletedEntity`")
        callback.onPostMigrate(db)
    }
}
